src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -304,34 +298,42 @@ private static bool CompilationHasOpenFileOnlyAnalyzers(CompilationWithAnalyzers
         {
             try
             {
-                // create result map
                 var version = await GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
 
+                var (fileLoadAnalysisResult, failedDocuments) = await GetDocumentLoadFailuresAsync(project, version, cancellationToken).ConfigureAwait(false);
```

Comment:
❕ Need to return `failedDocuments` to the pool after use. A reasonable alternative would be returning an `ImmutableHashSet<T>` instead, since this value will be empty in practice almost always and `ImmutableHashSet<T>.Empty` is a singleton.